### PR TITLE
New party refactor

### LINKED ISSUE #57: refactor hidden field to session in movie show through new party
validations that start time and date aren't in the past

### PR DESCRIPTION
**Pull request by catherinemdean15**
-update new party to use sessions instead of hidden params
-update session tests to use nav bar

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def create
       session[:user_id] = user.id
       redirect_to dashboard_user_path(user)
     else
-      flash[:errors] = user.errors.full_messages
+      flash[:notice] = 'Please fill in both an email and a password'
       flash[:notice] = "Passwords do not match" if !matching_password?
```
Comment (ninesky00) [inline app/controllers/users_controller.rb:15]: I'm not sure this is the best idea, this would leave out errors like already in database.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -24,7 +24,9 @@ def dashboard
   private
 
   def user_params
-    params.require(:user).permit(:email, :password, :password_confirmation)
+    params[:user][:email].downcase!
+    params.require(:user).permit(:email, :password)
+
```
Comment (ninesky00) [inline app/controllers/users_controller.rb:29]: does this work? might be easier to do this than the before save method I created

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (catherinemdean15) [reply app/controllers/users_controller.rb:15]: No errors were displaying while I was adjusting the tests for the nav bar, but I completely agree! We should look into this together.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (catherinemdean15) [reply app/controllers/users_controller.rb:29]: All tests were passing, but we should make sure we're testing all edge cases. The before save method was erroring with the validations in the test for some reason.
